test/version_test.exs

```diff
@@ -167,23 +167,32 @@ defmodule GitOps.Test.VersionTest do
   end
 
   test "last valid non rc is found correctly without prefixes" do
-    versions = ["0.0.1", "0.1.0-rc0", "0.1.0", "0.2.0-alpha"]
+    versions = ["0.0.1", "0.1.0-rc0", "0.1.0", "0.2.0-alpha", "0.11.0", "0.9.0"]
```

Comment:
This example is a bit confusing. Why would there be a `0.9.0` release *after* a `0.11.0` release?

Reply:
because that's how `git tag` returns it

Reply:
but git tag returns version tags applied in the order they were applied right? You wouldn't release a version 0.11.0 and then release a 0.9.0

Reply:
Specifically, in order of the commits that were tagged

Reply:
The point is that I don't think we want to make the new version greater than *any version that ever existed*. We'd want to increment the most recent appropriate version number.

Reply:
Nope, just tested 

```
git tag 0.9.0 
git tag 0.11.0 
```

but `git tag` still returns reverse order

Reply:
> We'd want to increment the most recent appropriate version number.
Not sure i understood what that means, but with current implementation i have the same changes in the changesets again and again, since it doesn't take the correct tag to compare with

Reply:
Hm...mine doesn't return them in reverse order. But if we can't rely on tags being in the right order for all versions of git, then that changes things quite a bit.

With that said, I was convinced a while ago that using git tags at all was a bad idea, and we should only be *making* tags, not reading them back. This is another argument in favor of that. This will break other repos that have (for whatever reason) had a higher version number in the past, but have since made a lower number version so we can't release this. What we can do, is fix the library so that it just uses that last version in `mix.exs` as its working point instead of git tags. THe only challenge will be that when a beta/rc series is started, we'll have to cache the last good major version somewhere. I'll put this on my list to handle soon.

Unfortunately, this would break current users workflows, and is just a bandaid to the bigger problem of using git tags at all so I can't merge this. My apologies. I'll fix the larger issue soon, though.